### ISSUE TITLE
Could not load devices error

### ISSUE THREAD

**Issue by 17tsejm1** [Hernas/homebridge-salus-sq610#10]
**Describe Your Problem:**
Just tried to add my gateway into Homebridge, seems like it's communicating but just unable to pull devices at all.

**Logs:**

[12/7/2024, 4:00:14 PM] [@hernas/homebridge-salus-sq610] Exception error:1C800064:Provider routines::bad decrypt, Error: error:1C800064:Provider routines::bad decrypt
[12/7/2024, 4:00:14 PM] [@hernas/homebridge-salus-sq610] Could not load the devices: Error: error:1C800064:Provider routines::bad decrypt
[12/7/2024, 4:00:14 PM] [@hernas/homebridge-salus-sq610] Could not load the devices: Error: error:1C800064:Provider routines::bad decrypt
[12/7/2024, 4:00:14 PM] [@hernas/homebridge-salus-sq610] Trying to fetch devices again in 60s...

**Plugin Config:**

{
    "ip_address": "192.168.40.13",
    "eu_id": "001E5E09027BEF5E",
    "refreshTime": 60,
    "_bridge": {
        "username": "0E:C7:AD:58:E5:B3",
        "port": 48471
    },
    "platform": "SalusSQ610HomebridgePlugin"
}

**Environment:**

* **Plugin Version**: 2.0.1
* **Homebridge Version**: 1.8.5
* **Node.js Version**: v22.12.0
* **NPM Version**: v10.9.2
* **Operating System**: macOS using hb-service

**Comment by 17tsejm1** [comment]
*Further retries:*

[12/7/2024, 4:29:42 PM] [@hernas/homebridge-salus-sq610] Exception error:1C800064:Provider routines::bad decrypt, Error: error:1C800064:Provider routines::bad decrypt
[12/7/2024, 4:29:42 PM] [@hernas/homebridge-salus-sq610] Could not load the devices: Error: error:1C800064:Provider routines::bad decrypt
[12/7/2024, 4:29:42 PM] [@hernas/homebridge-salus-sq610] Could not load the devices: Error: error:1C800064:Provider routines::bad decrypt
    at SalusConnect.makeEncryptedRequest (/Users/jordantse/.nvm/versions/node/v22.12.0/lib/node_modules/@hernas/homebridge-salus-sq610/src/SalusConnect.ts:144:15)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at SalusConnect.getDevices (/Users/jordantse/.nvm/versions/node/v22.12.0/lib/node_modules/@hernas/homebridge-salus-sq610/src/SalusConnect.ts:182:24)
    at SalusSQ610HomebridgePlatform.discoverDevices (/Users/jordantse/.nvm/versions/node/v22.12.0/lib/node_modules/@hernas/homebridge-salus-sq610/src/platform.ts:60:17)

**Comment by california444** [comment]
Hi @17tsejm1 ,
did you try to set EU-ID to 0000000000000000 ?
If not please try:
`"eu_id": "0000000000000000",`

**Comment by 17tsejm1** [comment]
@california444 changing the EU-ID has worked to get the devices added to HomeKit, but I seem to be unable to control any devices. There is a schedule enabled on the thermostat but the thermostat is currently in Off mode. Trying to change mode through HomeKit, it tries to set it then reverts immediately back to off mode.

**Comment by california444** [comment]
Changing modes was never enabled so far, if I am not mistaken. (With the option to enable cooling mode via hardware switch, this is very complicated to impossible)
you should be able to control the temperature and see the current temperature and humidity.

**Comment by 17tsejm1** [comment]
Appears it only works if the thermostat is in "Permanent Hold" mode and not "Follow Schedule". So must use the schedule from within HomeKit and not the thermostat.

Thanks.

**Comment by ipsumium** [comment]
Hello, I used the plugin before and it worked. Something happened and now i get a similar error:
[10/14/2025, 8:23:38 PM] [SalusSQ610HomebridgePlugin] Exception error:1C800064:Provider routines::bad decrypt, Error: error:1C800064:Provider routines::bad decrypt
[10/14/2025, 8:23:38 PM] [SalusSQ610HomebridgePlugin] Could not load the devices: Error: error:1C800064:Provider routines::bad decrypt
[10/14/2025, 8:23:38 PM] [SalusSQ610HomebridgePlugin] Could not load the devices: Error: error:1C800064:Provider routines::bad decrypt
    at SalusConnect.makeEncryptedRequest (/homebridge/node_modules/@hernas/homebridge-salus-sq610/src/SalusConnect.ts:144:15)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at SalusConnect.getDevices (/homebridge/node_modules/@hernas/homebridge-salus-sq610/src/SalusConnect.ts:182:24)
    at SalusSQ610HomebridgePlatform.discoverDevices (/homebridge/node_modules/@hernas/homebridge-salus-sq610/src/platform.ts:60:17)

I tried to set EU-ID to 0000000000000000 with no luck, i get the same error. I tried also to reinstall the plugin.

I have a sonoff ihost running homebridge in docker.
Plugin version v2.0.2
node -v  v22.20.0
node -p "process.versions.openssl"  3.5.2
